src/idl_gen_rust.cpp

```diff
@@ -1350,9 +1350,14 @@ class RustGenerator : public BaseGenerator {
         code_ +=
             "    if self.{{FIELD_TYPE_FIELD_NAME}}_type() == "
             "{{U_ELEMENT_ENUM_TYPE}} {";
-        code_ +=
+        if (field.required) {
```

Comment:
Would you please make a comment that this logic is not tested by the integration test, as of this date? Ideally, all code generator changes are accompanied by changes in the generated code. But, that's not the case here because we don't use this logic with the types specified in the FlatBuffers schemas for integration tests.